url9ba68db77a_samkio_Dungeonator_tgz-pJ8-net_woopa_dungeon_core_Main2J8/src/net/woopa/dungeon/core/CoreLevelCreator.java

```diff
@@ -0,0 +1,168 @@
+package net.woopa.dungeon.core;
+
+import java.util.ArrayList;
+
+import net.woopa.dungeon.core.CoreRoom.RoomType;
+import net.woopa.dungeon.datatypes.Direction;
+import net.woopa.dungeon.datatypes.Grid;
+import net.woopa.dungeon.datatypes.LevelCreator;
+import net.woopa.dungeon.datatypes.Vector2D;
+
+public class CoreLevelCreator implements LevelCreator {
+	private Vector2D levelStart, levelEnd;
+	private Direction startDir, endDir;
+	private ArrayList<CoreRoom> rooms = new ArrayList<CoreRoom>();
+
+	@Override
+	public Vector2D levelStart() {
+		return levelStart;
+	}
+
+	@Override
+	public Vector2D levelEnd() {
+		return levelEnd;
+	}
+
+	@Override
+	public Grid generate(Vector2D levelSize, Vector2D startLocation,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 38-92 into a separate method.